#1197 use octokit inside action to get commits

scottwn opened the pull request:
<!--
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Increment the version number in the /VERSION file. If this is a bugfix, increment the patch version (the rightmost number), for example 0.18.2 becomes 0.18.3. If it's a new feature, increment the minor version (the middle number), for example 0.18.2 becomes 0.19.0.
⚠️ Add a note to /CHANGELOG.md with your version number and the date, summarizing the changes.
⚠️ If your pull request fixes an open issue, please link to the issue.
⚠️ Rebase the latest changes from the develop branch. Assuming origin points to https://github.com/AgnostiqHQ/covalent, you should run git rebase origin/develop.
-->

- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AgnostiqHQ/covalent?pullRequest=1198) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AgnostiqHQ/covalent?pullRequest=1198) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: scottwn<br/>:x: CovalentOpsBot<br/><hr/>**CovalentOpsBot** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AgnostiqHQ/covalent?pullRequest=1198) it.</sub>

scottwn commented:
The code in `dist` is automatically generated and can be ignored.

In `action.yml` I've removed commits as an input since Node errors if there are too many commits being passed as an argument. I've added a token and a "basehead" formatted comparison so the action can retrieve commits on its own.

scottwn commented on .github/actions/changelog/index.js:
```diff
@@ -32,107 +33,131 @@ const get_author = (commit, authors) => {
+  const owner = github.context.payload.repository.organization
+  const repo = github.context.payload.repository.name
+  const octokit = github.getOctokit(token);
+  octokit.rest.repos
+    .compareCommitsWithBasehead({
+      owner: owner,
+      repo: repo,
+      basehead: basehead,
+    })
+    .then((compareCommits) => {
+      const {
+        data: { commits: commits },
+      } = compareCommits;
```
This is the only meaningful change to the JS. The rest of the diff is indentation and minor changes in variable names.

scottwn commented:
I've also made corresponding changes in `changelog.yml` to pass the correct vars to the action.
